backend/src/LibExecution/Interpreter.fs

```diff
@@ -829,26 +830,51 @@ and executeLambda
     // provide this error message here. We don't have this information in
     // other places, and the alternative is just to provide incompletes
     // with no context
-    let expectedLength = NEList.length l.parameters
-    let actualLength = NEList.length args
-    if expectedLength <> actualLength then
+
+    let paraList = parameters |> NEList.toList
+    let argsList = args |> NEList.toList
+
+    // Check destructuring
+    match argsList with
+    | [ DTuple(_, _, rest) ] when
```

Comment:
Can the destructuring be integrated with ELet destructuring and using the same code? It's not just about allowing tuples in lambdas, it's about having `let (x, y) =` do the same thing as `fun (x,y) -> `